pkg/component/runtime/service.go

```diff
@@ -452,19 +571,58 @@ func (s *serviceRuntime) install(ctx context.Context) error {
 func (s *serviceRuntime) uninstall(ctx context.Context) error {
 	// Always retry for internal attempts to uninstall, because they are an attempt to converge the agent's current state
 	// with its desired state based on the agent policy.
-	return uninstallService(ctx, s.log, s.comp, s.executeServiceCommandImpl, true)
+	return uninstallService(ctx, s.log, s.comp, "", s.executeServiceCommandImpl, true)
```

Comment:
This will currently retry forever on failure, and after https://github.com/elastic/elastic-agent/pull/3086 it will retry for up to 1 hour.

You should mark signature validation as a permanent failure to short circuit the retry logic: https://pkg.go.dev/github.com/cenkalti/backoff/v4#Permanent

This can be done everywhere a signature failure could be retried.